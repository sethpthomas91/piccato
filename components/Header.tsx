import styles from '../styles/header.module.css';
import Router from 'next/router';
import Head from 'next/head'

const Header = () => {
  return (
    <nav className={styles.header}>
      <Head>
        <title>Pic-Cat-O</title>
        <meta name="description" content="where people go to get catified" />
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <h1 className={styles.title} 
        data-testid="titlePresser"
        onClick={() => Router.push('/')}>
          PIC-CAT-TO
      </h1>
      <a data-testid="createPresser" onClick={() => Router.push('/create/').then(()=>{Router.reload()})}>
        <img className={styles.icon} src='/create_icon.jpg'/>
      </a>
    </nav>
  );
}

export default Header
