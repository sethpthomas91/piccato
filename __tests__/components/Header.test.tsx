import renderer from 'react-test-renderer';
import Header from '../../components/Header';
import '@testing-library/jest-dom';
import { render, act, fireEvent, screen, waitFor, getByTestId } from '@testing-library/react';
import mockRouter from 'next-router-mock';
import singletonRouter from 'next/router';

jest.mock('next/router', () => require('next-router-mock'));
jest.mock('next/dist/client/router', () => require('next-router-mock'));

describe ('Header', () => {

  it('renders correctly', () => {
    const tree = renderer
      .create(<Header/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  beforeEach(() => {
    mockRouter.setCurrentUrl("/");
  });


  it('pushes to root url', () => {
    render(
      <Header/>
    );
    fireEvent.click(screen.getByTestId('titlePresser'));
    expect(singletonRouter).toMatchObject({ asPath: '/' });
  });

  it('pushes to create url', () => {
    render(
      <Header/>
    );
    fireEvent.click(screen.getByTestId('createPresser'));
    expect(singletonRouter).toMatchObject({ asPath: '/create' });
  });
})
