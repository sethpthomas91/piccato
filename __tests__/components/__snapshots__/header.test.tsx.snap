// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<nav
  className="header"
>
  <h1
    className="title"
    data-testid="titlePresser"
    onClick={[Function]}
  >
    PIC-CAT-TO
  </h1>
  <a
    data-testid="createPresser"
    onClick={[Function]}
  >
    <img
      className="icon"
      src="/create_icon.jpg"
    />
  </a>
</nav>
`;
